pkg/server/backend_manager.go

```diff
@@ -97,6 +100,22 @@ func (b *backend) Send(p *client.Packet) error {
 	return err
 }
 
+func (b *backend) Recv() (*client.Packet, error) {
+	b.recvLock.Lock()
+	defer b.recvLock.Unlock()
+
+	const segment = commonmetrics.SegmentFromAgent
+	pkt, err := b.conn.Recv()
+	if err != nil && err != io.EOF {
+		metrics.Metrics.ObserveStreamErrorNoPacket(segment, err)
+	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nit: this would be a bit easier to follow up it were restructured like this:
```suggestion
	if err != nil {
		if err != io.EOF {
			metrics.Metrics.ObserveStreamErrorNoPacket(segment, err)
		}
		return nil, err
	}
```

Reply:
Done, including elsewhere.